frontend/components/MainPage/sections/RewardsSection/StakingRewardsThisEpoch.tsx

```diff
@@ -0,0 +1,81 @@
+import { InfoCircleOutlined } from '@ant-design/icons';
+import { useQuery } from '@tanstack/react-query';
+import { Popover, Typography } from 'antd';
+import { gql, request } from 'graphql-request';
+import { useMemo } from 'react';
+import { z } from 'zod';
+
+import { SUBGRAPH_URL } from '@/constants/urls';
+import { useStakingProgram } from '@/hooks/useStakingProgram';
+import { formatToTime } from '@/utils/time';
+
+const { Text } = Typography;
+
+const EpochTimeResponseSchema = z.object({
+  epochLength: z.string(),
+  blockTimestamp: z.string(),
+});
+type EpochTimeResponse = z.infer<typeof EpochTimeResponseSchema>;
+
+const useEpochEndTime = () => {
+  const { activeStakingProgramAddress } = useStakingProgram();
+  const latestEpochTimeQuery = useMemo(() => {
```

Comment:
can just be a string outside of this hook, you use `enabled: !!activeStakingProgramAddress,` anyway in useQuery